flang/tools/tco/tco.cpp

```diff
@@ -85,7 +85,9 @@ static int compileFIR() {
   fir::KindMapping kindMap{context.get()};
   mlir::PassManager pm{context.get()};
   mlir::applyPassManagerCLOptions(pm);
-  if (emitFir) {
+  if (passPipeline.hasAnyOccurrences()) {
+    passPipeline.addToPipeline(pm);
+  } else if (emitFir) {
     // parse the input and pretty-print it back out
     // -emit-fir intentionally disables all the passes
```

Comment:
Is "-emit-fir intentionally disables all the passes" still true after this change ?  If not, maybe it could be updated.

Reply:
Oh okay, I will switch the conditions.